### ISSUE TITLE
span/5 supporting difference list

### ISSUE THREAD

**Issue by mndrix** [mndrix/list_util#22]
A version of span/4 which supports a difference list like:

``` prolog
?- span(==(a), [a,a,b,c,a], Prefix, Tail, Suffix).
Prefix = [a, a|Tail],
Suffix = [b, c, a].
```

**Comment by eazar001** [comment]
There are a few errant test cases in this. I'll push the corrections in the next commit.

**Comment by eazar001** [comment]
The latest fixes, should represent a functional `span/5`.

**Comment by mndrix** [comment]
@eazar001 Thanks for the great work here.  The test cases are especially thorough.  I pushed a change in commit eef5ea71103608dc1d897ddc0c841ca5a9423864 to adjust the semantics for cases where span/5 finds 0 matching elements.  Your first commit in the series actually had the right semantics.

Basically, it's because of how difference lists work.  The list `[a,b,c]` is represented as a difference as `[a,b,c|Tail]-Tail`.  That's because the difference between `[a,b,c|Tail]` and `Tail` is `[a,b,c]`.  By extension the list `[]` is represented as a difference list `Tail-Tail` since the difference between `Tail` and `Tail` is nothing.

If this all makes sense to you, I'll publish a new version tomorrow with all these changes.

For convenience, here is a [summary of changes for span/5 so far](https://github.com/mndrix/list_util/compare/v0.11.0...eef5ea71103608dc1d897ddc0c841ca5a9423864).

**Comment by eazar001** [comment]
Okay I see, the empty list succeeding does make sense when juxtaposed with your explanation now. Thank you for clearing this up for me.

**Comment by mndrix** [comment]
Released as part of v0.12.0.  Thanks again @eazar001

**Comment by eazar001** [comment]
Pleasure as always, @mndrix
